analysis_service/base/models.py

```diff
@@ -0,0 +1,206 @@
+from datetime import datetime, timedelta
+from pytz import UTC
+from django.db import models
+from django.contrib.auth.models import User
+
+from analysis_service.base.util import provisioning, scheduling
+
+
+class Cluster(models.Model):
```

Comment:
> Do you think we should enforce that, for example, the identifiers must be unique for clusters?
I think we should either s/identifier/name or make the constraint more explicit. What is the identifier string used for?